tests/test_deployment_configuration.py

```diff
@@ -0,0 +1,133 @@
+                                                                                                         operator="NotIn",
+                                                                                                         values=["b"])])))]
+
+    pod_anti_affinity = PodAntiAffinity(required_during_scheduling_ignored_during_execution=pod_anti_affinity_terms,
+                                        preferred_during_scheduling_ignored_during_execution=wpaats)
+
+    psts = [PreferredSchedulingTerm(weight=100, preference=NodeSelectorTerm(
+        match_expressions=[NodeSelectorRequirement(key="exp2", operator="NotIn", values=["aaaa", "bvzv", "czxc"])],
+        match_fields=[NodeSelectorRequirement(key="fields3", operator="NotIn", values=["aaa", "cccc", "zxcc"])]))]
+    node_selector = NodeSelector(node_selector_terms=[
+        NodeSelectorTerm(match_expressions=[NodeSelectorRequirement(key="exp1", operator="Exists")],
+                         match_fields=[NodeSelectorRequirement(key="fields1", operator="Exists")])])
+    node_affinity = NodeAffinity(required_during_scheduling_ignored_during_execution=node_selector,
+                                 preferred_during_scheduling_ignored_during_execution=psts)
+
+    affinity = Affinity(pod_affinity=pod_affinity, pod_anti_affinity=pod_anti_affinity, node_affinity=node_affinity)
+
+    config_builder = DeploymentConfigurationBuilder(name="cool-deployment-config", cluster=None)
+    config_builder.with_hpa(max_replicas=10, min_replicas=2, target_cpu_utilization_percentage=80). \
+        with_pod_node_selector({"im": "a map", "foo": "bar"}). \
+        with_resource_requirements(limits={"cpu": "4", "memory": "4g"}, requests={"cpu": "2", "memory": "2g"}). \
+        with_replicas(replica_count=4). \
+        with_toleration(effect="PreferNoSchedule", key="equalToleration", toleration_seconds=30, operator="Equal", value="kek"). \
+        with_toleration(effect="PreferNoSchedule", key="equalToleration", toleration_seconds=30, operator="Exists"). \
+        with_affinity(affinity)
+
+    config_builder.build = MagicMock(return_value=DeploymentConfiguration(name=config_builder.name,
+                                                                          hpa=config_builder.hpa,
+                                                                          pod=config_builder.pod_spec,
+                                                                          container=config_builder.container_spec,
+                                                                          deployment=config_builder.deployment_spec))
+
+    new_config = config_builder.build()
+    camel_case_config = new_config.to_camel_case_dict()
+
+    print(json.dumps(deployment_config_json, sort_keys=True))
+    print(json.dumps(camel_case_config, sort_keys=True))
+
+    assert json.dumps(camel_case_config, sort_keys=True) == json.dumps(deployment_config_json, sort_keys=True)
```

Comment:
Should we test GET/POST/DELETE requests as we do with other entities?

Reply:
Yes, we should. This PR is still a work-in-progress. I'll add tests after I get my hands on working example of `hydro-serving-manager`